### DISCUSSION TITLE
アプリ使用中に、連携済みのVRoidHub連携画面を解除した後、再度表示させる方法

### DISCUSSION THREAD

**Discussion by wcl1207** [pixiv/vroid-sdk-developers#146] [Q&A]
いつもお世話になっております。

アプリで一回VRoidHub連携を実施しました。その後、アプリ使用中に、別途にhttps://hub.vroid.com/settings/authorized_apps で連携解除を実施すると、エラーが出て、VRoidHub連携画面が再びでることはできず、VRoidHubでの他の操作ができなくなりました。
そのため、一回VRoidHub連携を実施した後、**アプリ使用中に**、VRoidHub連携を解除した後、もう一回VRoidHub連携画面が出ることができるように、どのAPIを呼び出して良いでしょうか。そのとも、他に良い方法でしょうか。

エラーメッセージを連携いたします。
>[VRoid SDK] ApiErrorFormat: {code:"AUTHORIZED_ERROR", message:"{"error":"invalid_grant","error_description":"指定された認可グラントは無効か、有効期限切れか、リダイレクトURIが異なるか、もしくは別のクライアントに適用されています。"}"}
UnityEngine.Debug:LogError (object)
Pixiv.VroidSdk.Unity.Logger.UnityLogWriter:LogError (object)
Pixiv.VroidSdk.Logger.SdkLogger:LogError (object)
VRoidSDK.Examples.Core.Renderer.ApiErrorRenderer:.ctor (VRoidSDK.Examples.Core.Model.ApplicationModel) (at Assets/VRoidSDK/Examples/Core/Renderer/ApiErrorRenderer.cs:15)
VRoidSDK.Examples.CharacterModelExample.Controller.CharacterModelsController/<>c__DisplayClass9_0:<GetOnErrorCallback>b__0 (Pixiv.VroidSdk.Api.DataModel.ApiErrorFormat) (at Assets/VRoidSDK/Examples/CharacterModelExample/Scripts/Controller/CharacterModelsController.cs:110)
Pixiv.VroidSdk.Utility.ResponseUtil/<>c__DisplayClass1_0:<OnHttpRequestFailed>b__1 (System.Exception)
Pixiv.VroidSdk.Oauth.Strategy.ForceRefresh:<GetTokenAsync>b__10_1 (Pixiv.VroidSdk.Networking.HttpRequestFailedException)
Pixiv.VroidSdk.Networking.Connections.HttpClientConnection/<>c__DisplayClass33_2:<SetResult>b__2 (object)
UnityEngine.UnitySynchronizationContext:ExecuteTasks () (at /Users/bokken/build/output/unity/unity/Runtime/Export/Scripting/UnitySynchronizationContext.cs:107)

**Comment by ytomi-pixiv** [comment]
いただいた状況ですが、VRoid Hubの方で連携が外れているためにRefreshTokenの発行に失敗する状況かと思います。SDK側でRevoke (access_token.json)削除すれば再連携可能になるかと考えられます。
`Pixiv.VroidSdk.Oauth.Client` の `ReleaseAuthorizedAccount` を呼び出し再連携できるか試していただけますでしょうか。
Exampleですと `ApiController#Logout` の処理にあたります。

[https://developer.vroid.com/sdk/docs/0.5.1/Pixiv.VroidSdk.Oauth.Client.html#Pixiv_VroidSdk_Oauth_Client_ReleaseAuthorizedAccount](https://developer.vroid.com/sdk/docs/0.5.1/Pixiv.VroidSdk.Oauth.Client.html#Pixiv_VroidSdk_Oauth_Client_ReleaseAuthorizedAccount)

**Reply by wcl1207** [reply]
ありがとうございます！確認させていただきます。
